internal/controller/postgrescluster/snapshots.go

```diff
@@ -133,49 +148,292 @@ func (r *Reconciler) reconcileVolumeSnapshots(ctx context.Context,
+	pvc *corev1.PersistentVolumeClaim,
+) (*corev1.PersistentVolumeClaim, error) {
+	var err error
+
+	// An InstanceSet must be chosen to scale resources for the dedicated snapshot volume.
+	// TODO: We've chosen the first InstanceSet for the time being, but might want to consider
+	// making the choice configurable.
+	instanceSpec := cluster.Spec.InstanceSets[0]
+
+	pvc.Annotations = naming.Merge(
+		cluster.Spec.Metadata.GetAnnotationsOrNil(),
+		instanceSpec.Metadata.GetAnnotationsOrNil())
+
+	pvc.Labels = naming.Merge(
+		cluster.Spec.Metadata.GetLabelsOrNil(),
+		instanceSpec.Metadata.GetLabelsOrNil(),
+		labelMap,
+	)
+
+	err = errors.WithStack(r.setControllerReference(cluster, pvc))
+	if err != nil {
+		return pvc, err
+	}
+
+	pvc.Spec = instanceSpec.DataVolumeClaimSpec
+
+	// Set the snapshot volume to the same size as the pgdata volume. The size should scale with auto-grow.
+	r.setVolumeSize(ctx, cluster, pvc, instanceSpec.Name)
+
+	// Clear any set limit before applying PVC. This is needed to allow the limit
+	// value to change later.
+	pvc.Spec.Resources.Limits = nil
+
+	err = r.handlePersistentVolumeClaimError(cluster,
+		errors.WithStack(r.apply(ctx, pvc)))
+	if err != nil {
+		return pvc, err
+	}
+
+	return pvc, err
+}
+
+// dedicatedSnapshotVolumeRestore creates a Job that performs a restore into the dedicated
+// snapshot volume.
+// This function is very similar to reconcileRestoreJob, but specifically tailored to the
+// dedicated snapshot volume.
+func (r *Reconciler) dedicatedSnapshotVolumeRestore(ctx context.Context,
+	cluster *v1beta1.PostgresCluster, dedicatedSnapshotVolume *corev1.PersistentVolumeClaim,
+	instances *observedInstances, clusterVolumes []corev1.PersistentVolumeClaim,
+	backupJob *batchv1.Job,
+) error {
+
+	pgdata := postgres.DataDirectory(cluster)
+	repoName := backupJob.GetLabels()[naming.LabelPGBackRestRepo]
+
+	opts := []string{
+		"--stanza=" + pgbackrest.DefaultStanzaName,
+		"--pg1-path=" + pgdata,
+		"--repo=" + regexRepoIndex.FindString(repoName),
+		"--delta",
+		// "--link-all",
+	}
+
+	// NOTE: It seems like our current architecture is forcing us to care about the instanceSet/instance
+	// and consistently use the same one as the user *could* have multiple instanceSets with different
+	// storage settings. Do we want to simply always choose the first instanceSet and instance in their
+	// respective arrays? Or do we want to add something to the API for the user to specify an instanceSet?
+	// For now, I am choosing the first instanceSet/instance in their arrays.
+	instanceSet := &cluster.Spec.InstanceSets[0]
+	instance := instances.bySet[instanceSet.Name][0].Runner
+
+	// How are we going to handle tablespace volumes? Feels like if they exist we need to provide them to
+	// the restore command otherwise they will break in any clones made from a snapshot.
+	pgtablespaces, err := r.reconcileTablespaceVolumes(ctx, cluster, instanceSet, instance, clusterVolumes)
```

Comment:
❓ What is returned here? Is this the regular set of tablespace volumes? Do we need dedicated copies to quiesce and snapshot?